mobile/lib/widgets/layout/main_navigation_drawer.dart

```diff
@@ -59,11 +65,106 @@ class MainNavigationDrawer extends StatelessWidget {
           (DrawerDestination drawerDestination) =>
               drawerDestination.destination,
         ),
-        const Padding(
-          padding: EdgeInsets.fromLTRB(28, 16, 28, 10),
+        const DrawerItem(
           child: Divider(),
         ),
+        const DrawerItem(
+          child: UserProfile(),
+        ),
+        const DrawerItem(
+          child: AuthButton(),
+        ),
       ],
     );
   }
 }
+
+class DrawerItem extends StatelessWidget {
+  final Widget child;
+  const DrawerItem({super.key, required this.child});
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+        padding: const EdgeInsets.fromLTRB(28, 16, 28, 10), child: child);
+  }
+}
+
+/// Display the user's profile information.
+class UserProfile extends StatelessWidget {
+  const UserProfile({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final User? user = context.watch<AuthState>().user;
+    if (user != null) {
+      return Row(
+        children: <Widget>[
+          SquareAvatar(
+            size: 45,
+            avatar: Image.network(user.avatarUrl),
+          ),
+          const SizedBox(width: 16),
+          Text(
+            user.username,
+            style: Theme.of(context).textTheme.titleLarge,
+          ),
+        ],
+      );
+    } else {
+      return Container();
+    }
```

Comment:
```suggestion
    }
    
    return Container();
```

On est sûr de vouloir mettre un container vide ?

Reply:
J'aurais préféré mettre rien ou un Widget qui fait office de vide